### DISCUSSION TITLE
How to set the package installation directory to the runtime shared library (.so files) search paths?

### DISCUSSION THREAD

**Discussion by Gwill** [pybind/pybind11#3507] [Q&A]
How to set the package installation directory ( python-path/lib/site-packages/package_name ) to the runtime shared library (.so files) search paths?

here is my config of Pybind11Extension in setup.py:

```
ext_modules = [
    Pybind11Extension(
        "package_name._core",
        [
            "src/main.cpp",
        ],
        include_dirs=["include"],
        library_dirs=["package_name"],
        libraries=["aaa", "bbb"],
        compiler_flags = [
            "-std=c++17",  # standard
            "-O3",  # Optimization
            "-Wno-delete-incomplete", "-Wno-sign-compare",
        ],
        runtime_library_dirs = ["$ORIGIN"]
    ),
]
```

I have set the runtime_library_dirs = ["$ORIGIN"]，but it cant find so file, how to set the package installation directory ( python-path/lib/site-packages/package_name ) to the runtime shared library (.so files) search paths?

**Comment by matte86** [comment]
I also have a similar doubt when building my project under MacOS.
My python binding depends on a shared object (.dylib) that is built with conan / cmake. On top of this I have my Python interface.
Everything seems to work as long as I am in my build tree but when I package everything so that I can distribute it and retrieve it with pip I have errors due to python not being able to find my library when importing the module.
I think this is something related to RPATH handling but I am not sure how to take it from here.
I know that cmake offers install_name_tool routines to fix this but I am not a python expert so I am not entirely sure how to integrate this into the setup.py and at which level.
